### PR TITLE
Adds generator to run wasm_spec_tests as EOSIO unit tests

### PR DESCRIPTION
**Pull request by jeffreyssmith2nd**
Adds the C++ and Python code used to generate the tests.

Also adds the hand edited wasms and notes on what was edited and why.

### CODE REVIEW

File: tests/eosio_spec_test_generator/altered-wasms/memory_size/notes.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## Changes
```
Comment (spoonincode) [inline tests/eosio_spec_test_generator/altered-wasms/memory_size/notes.md]: Why did we make these changes?

Comment (jeffreyssmith2nd) [reply tests/eosio_spec_test_generator/altered-wasms/memory_size/notes.md]: When we build the unit test, the `eosio::check` messages are added to the end of the WASM as data segments. So we need at least one page of memory or we get an error for the data segment being out of memory.

**Comment by spoonincode** [comment]
I'm a little confused how tests are included vs not included. For example, take call_indirect.wast in the spec tests, this has dozens of tests. But the resulting test in EOSIO/eos#7784 only has 17 tests. Actually it's somewhat interesting that call_indirect.wast has 15 assert_trap/assert_exhaustion and call_indirect.0.cpp also appears to have 15 failures cases. But where did all the other assert_returns go to -- call_indirect.0.cpp only has two expected passing tests.

**Comment by jeffreyssmith2nd** [comment]
I’ll update this PR with a readme that explains some of the logic since it’s not immediately clear from the code.

But the short answer is every up to 100 assert_return tests are grouped into one unit test.

**Comment by jeffreyssmith2nd** [comment]
@spoonincode Added a commit with a much more thorough explanation of the process and the logic behind some of the choices that were made.

Let me know if there's more info needed/anything is unclear and I can attempt to improve it.

**Comment by jeffreyssmith2nd** [comment]
> > > * I'm not sure that this belongs in eos-vm.  I'm uncomfortable with having eos-vm have a dependency on cdt.
> > 
> > 
> > It started out here because the test wasms were directly in this repo. I'm certainly not opposed to moving it, though I'm not sure where the most logical place for it would be.
> 
> I would be for moving this to the either eos-vm-test-wasms or eos itself. eos already has a light coupling to CDT for test contracts anyway so it would be a reasonable home for this.

Todd mentioned concerns in the demo about adding a large number of binary files to the eos repo and suggested putting the generated cpp/wasms in a new repo that is submoduled into eos. That may be a logical place to put the generator code as well.

**Comment by jeffreyssmith2nd** [comment]
Closing because its been moved to a new repo

**Pull request closed without merging**